src/bin/am/commands/start.rs

```diff
@@ -134,44 +134,43 @@ impl Arguments {
 
 #[derive(Debug, Clone)]
 struct Endpoint {
-    pub url: url::Url,
-    pub job_name: String,
+    url: Url,
+    job_name: String,
 }
 
 impl Endpoint {
-    fn new(url: url::Url, job_name: String) -> Self {
+    fn new(url: Url, job_name: String) -> Self {
         Self { url, job_name }
     }
-}
 
-impl From<Endpoint> for prometheus::ScrapeConfig {
     /// Convert an InnerEndpoint to a Prometheus ScrapeConfig.
     ///
     /// Scrape config only supports http and https atm.
-    fn from(endpoint: Endpoint) -> Self {
-        let scheme = match endpoint.url.scheme() {
+    fn into_scrape_config(self, pushgateway_enabled: bool) -> prometheus::ScrapeConfig {
```

Comment:
Maybe just call this `honor_labels` since that is what it controls:

```suggestion
    fn into_scrape_config(self, honor_labels: bool) -> prometheus::ScrapeConfig {
```